File: problems/1030-matrix-cells-in-distance-order/solution.cpp
class Solution {
public:
    vector<vector<int>> allCellsDistOrder(int rows, int cols, int rCenter, int cCenter) {
        vector<vector<int>> result;
        vector<pair<int, vector<int>>> distances;

        for (int i = 0; i < rows; ++i) {
            for (int j = 0; j < cols; ++j) {
                int distance = abs(i - rCenter) + abs(j - cCenter);
                distances.push_back({distance, {i, j}});
            }
        }

        sort(distances.begin(), distances.end());

        for (const auto& p : distances) {
            result.push_back(p.second);
        }

        return result;
    }
};

File: problems/1030-matrix-cells-in-distance-order/solution.java
import java.util.Arrays;

class Solution {
    public int[][] allCellsDistOrder(int rows, int cols, int rCenter, int cCenter) {
        int[][] result = new int[rows * cols][2];
        int index = 0;
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                result[index][0] = i;
                result[index][1] = j;
                index++;
            }
        }

        Arrays.sort(result, (a, b) -> {
            int distA = Math.abs(a[0] - rCenter) + Math.abs(a[1] - cCenter);
            int distB = Math.abs(b[0] - rCenter) + Math.abs(b[1] - cCenter);
            return distA - distB;
        });

        return result;
    }
}